src/perf/java/rx/observables/SyncOnSubscribePerf.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.observables;
+
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+
+import rx.Observable.OnSubscribe;
+import rx.Observable;
+import rx.Subscriber;
+import rx.internal.operators.OnSubscribeFromIterable;
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(Scope.Thread)
+public class SyncOnSubscribePerf {
+
+    public static void main(String[] args) {
+        SingleInput singleInput = new SingleInput();
+        singleInput.size = 1;
+        singleInput.setup(generated._jmh_tryInit_());
+        SyncOnSubscribePerf perf = new SyncOnSubscribePerf();
+        perf.benchSyncOnSubscribe(singleInput);
+    }
+    private static class generated {
+        private static Blackhole _jmh_tryInit_() {
+            return new Blackhole();
+        }
+    }
+    
+    private static SyncOnSubscribe<Void, Integer> createSyncOnSubscribe(final Iterator<Integer> iterator) {
+        return new SyncOnSubscribe<Void, Integer>() {
+            @Override
+            protected Void next(Void state, Subscriber<? super Integer> subscriber) {
+                if (iterator.hasNext())
+                    subscriber.onNext(iterator.next());
+                else
+                    subscriber.onCompleted();
+                return null;
+            }};
+    }
+    
+    @Benchmark
+//  @Group("single")
+    public void benchSyncOnSubscribe(final SingleInput input) {
+        createSyncOnSubscribe(input.iterator).call(input.newSubscriber());
```

Comment:
The problem with this is that the iterator gets consumed once and you then benchmark how fast an empty iterator comes to completion through the operator.